src/model/page-config-resolver.js

```diff
@@ -266,17 +266,18 @@ class JsonLdParser {
       possibleConfigs = [possibleConfigs];
     }
 
-    for (let i = 0; i < /** @type {Array} */ (possibleConfigs).length; i++) {
-      const possibleConfig = possibleConfigs[i];
+    const configs = /** @type {!Array<!JsonObject>} */ (possibleConfigs);
+    for (let i = 0; i < configs.length; i++) {
+      const config = configs[i];
 
       // Must be an ALLOWED_TYPE
-      if (!this.checkType_.checkValue(possibleConfig['@type'], ALLOWED_TYPES)) {
+      if (!this.checkType_.checkValue(config['@type'], ALLOWED_TYPES)) {
         continue;
       }
 
       // Must have a isPartOf[@type=Product].
       let productId = null;
-      const partOfArray = this.valueArray_(possibleConfig, 'isPartOf');
+      const partOfArray = this.valueArray_(config, 'isPartOf');
       if (partOfArray) {
         for (let i = 0; i < partOfArray.length; i++) {
```

Comment:
Do you want to use a different for loop for the overload of the `i` variable here? Or maybe just use a `j` instead?

Reply:
Renaming to `j` sounds good. Just a note, the inner loop's `i` variable wasn't rewriting the outer loop's `i`

Reply:
The for-of loop would have rocked, but we currently can't use those in JS that gets compiled into binaries :sob: . The other loop suggestions seemed like downgrades to me, in terms of readability and/or efficiency